source/opt/loop_dependence.cpp

```diff
@@ -883,5 +1111,524 @@ PartitionedSubscripts LoopDependenceAnalysis::PartitionSubscripts(
   return partitions;
 }
 
+bool IsConstant(const SENode* node) {
```

Comment:
enclose in anonymous namespace, please.  Or, better yet, make this an class function of SENode?

Reply:
I've removed it altogether and replaced the uses with `SENode::AsSEConstantNode`